app/supplejack/extraction/concerns/enrichment_extraction_process.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Extraction
+  module Concerns
+    module EnrichmentExtractionProcess
+      extend ActiveSupport::Concern
+
+      included do
+        def process_enrichment_extraction(enrichment_params)
+          parsed_params = JSON.parse(enrichment_params)
+          extraction_context = build_extraction_context(parsed_params)
+
+          enrichment_extraction = extraction_context.enrichment_extraction
+          return unless enrichment_extraction.valid?
+
+          enrichment_extraction.extract_and_save
+          handle_harvest_job(extraction_context)
+        end
+      end
+
+      private
+
+      ExtractionContext = Struct.new(
```

Comment:
Is this also a value object? I feel like it would be good to define it outside of this class.